src/spi.rs

```diff
@@ -1096,6 +1102,77 @@ where
     }
 }
 
+pub struct DmaBuilder<SPI> {
+    spi: SPI,
+}
+
+pub struct Tx<SPI> {
+    spi: PhantomData<SPI>,
+    dr: u32,
+}
+
+pub struct Rx<SPI> {
+    spi: PhantomData<SPI>,
+    dr: u32,
+}
```

Comment:
If you bound this `SPI` to an `Instance` trait (or other name) with a `fn dr() -> u32` method, you could remove this `dr` field and save some ram. You could then implement this `Instance` trait for `SPI1` (and others) with something like:
```
unsafe { &(&*SP1::ptr().dr) as *const _ as u32 }
```

Reply:
Sounds good.